neqo-crypto/src/agent.rs

```diff
@@ -50,6 +50,110 @@ use crate::{
     time::{Time, TimeHolder},
 };
 
+/// Private trait for Certificate Compression implementation
+/// # Safety
+///
+/// Use `SafeCertCompression` to implement an encoder/decoder instead.
+unsafe trait UnsafeCertCompression {
+    unsafe extern "C" fn decode_callback(
+        input: *const ssl::SECItem,
+        output: *mut ::std::os::raw::c_uchar,
+        output_len: usize,
+        used_len: *mut usize,
+    ) -> ssl::SECStatus;
+
+    unsafe extern "C" fn encode_callback(
+        input: *const ssl::SECItem,
+        output: *mut ssl::SECItem,
+    ) -> ssl::SECStatus;
+}
+
+/// The trait is used to represent a certificate compression data structure
+/// Used in order to enable Certificate Compression extension during TLS connection
+pub trait SafeCertificateCompression {
+    /// Certificate Compression identifier as in RFC8879
+    const ID: u16;
+    /// Certification Compression name (used only for logging/debugging)
```

Comment:
This isn't used for reporting?  Seems like that could be improved in NSS.

Reply:
It's used for tracing, like we will log the message like "Server uses Zlib (for example) for decoding the received certificate"

Reply:
See above.  We can do better (not for this change necessarily).